app/Services/User/StorageService.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace App\Services\User;
+
+use App\Exceptions\ApiException;
+use App\Exceptions\ExceptionFactory;
+use App\Exceptions\FatalExceptionFactory;
+use App\Models\User;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\DB;
+use JsonException;
+use Throwable;
+
+class StorageService
+{
+    /** @throws JsonException */
+    private function getUserStorage(User $user): array
+    {
+        return json_decode(
+            DB::table('users')->find($user->id, 'storage')->storage ?? '[]',
+            associative: true,
+            flags: JSON_THROW_ON_ERROR,
+        );
+    }
+
+    private function setUserStorage(User $user, array $storage): void
+    {
+        DB::table('users')->where('id', $user->id)->update([
+            'storage' => json_encode($storage),
+        ]);
+    }
+
+    private function getStorageKey(array $storage, string $key): JsonResponse
+    {
+        return new JsonResponse(
+            ($key === '')
+                ? json_encode(array_keys($storage))
```

Comment:
Ten ternary powinien być w get poniżej i wołać getStorageKey albo getKeys. A już getStorageKey to powinno robić to co nazwa mówi.

Zresztą pusty string jest poprawnym kluczem JSONowym, no ale i tak nie będzie możliwości zapisania pod nim, no trudno.